projects/packages/jetpack-mu-wpcom/src/features/code-editor/code-editor/code-editor.tsx

```diff
@@ -1,402 +1,174 @@
+  visibility: hidden;
+}`;
+	document.head.appendChild( styleElement );
 }
 
-/**
- * Render the code editor
- *
- * @return The code editor element.
- */
-function CodeEditor(): JSX.Element {
-	const instanceId = useInstanceId( CodeEditor );
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	const { content, blocks, type, id } = useSelect( ( select: any ) => {
-		const { getEditedEntityRecord } = select( coreStore );
-		const { getCurrentPostType, getCurrentPostId } = select( editorStore );
-		const _type = getCurrentPostType();
-		const _id = getCurrentPostId();
-		const editedRecord = getEditedEntityRecord( 'postType', _type, _id );
-
-		return {
-			content: editedRecord?.content,
-			blocks: editedRecord?.blocks,
-			type: _type,
-			id: _id,
-		};
-	}, [] );
-
-	// Replicates the logic found in getEditedPostContent().
-	const value = React.useMemo( () => {
-		if ( content instanceof Function ) {
-			return content( { blocks } );
-		} else if ( blocks ) {
-			// If we have parsed blocks already, they should be our source of truth.
-			// Parsing applies block deprecations and legacy block conversions that
-			// unparsed content will not have.
-			return __unstableSerializeAndClean( blocks );
-		}
-		return content;
-	}, [ content, blocks ] );
-
-	return (
-		<>
-			<VisuallyHidden as="label" htmlFor={ `post-content-${ instanceId }` }>
-				{ __( 'Type text or HTML', 'jetpack-mu-wpcom' ) }
-			</VisuallyHidden>
-			<CM initialValue={ value } type={ type } id={ id } />
-		</>
+const observer = new MutationObserver( () => {
+	const codeEditorTextarea: ReactHTMLTextAreaElement | null = document.querySelector(
+		codeEditorTextareaSelector
 	);
-}
-
-const cm_lazy = ( cm_module: typeof import('@a8cCodeEditor/codemirror-bundle') ) => {
-	return function CodeMirror( props: { initialValue: string; type: string; id: string } ) {
-		const {
-			Autocomplete,
-			Commands,
-			HtmlLanguage,
-			Language,
-			Lint,
-			Search,
-			State,
-			View,
-			syntaxHighlightingStyle,
-			theme,
-		} = cm_module;
-
-		const { initialValue, type, id } = props;
-
-		const ref: React.RefObject< HTMLDivElement > = React.useRef( null );
 
-		const viewRef: React.MutableRefObject< EditorView | undefined > = React.useRef( undefined );
-
-		const { editEntityRecord } = useDispatch( coreStore );
-		const updateCode = React.useCallback(
-			( code: string ) => {
-				editEntityRecord( 'postType', type, id, {
-					content: code,
-					blocks: undefined,
-					selection: undefined,
-				} );
-			},
-			[ type, id, editEntityRecord ]
-		);
-
-		const updateListener = React.useRef(
+	// If there's a textarea
+	if ( codeEditorTextarea ) {
+		// And the editor isn't loaded or initializing
+		if ( ! editor ) {
+			// Do it
+			setupEditor( codeEditorTextarea );
+		}
+	}
+	// If there's no textarea but the editor is loaded
+	// we should "unmount" it.
+	else if ( editor && editor !== true ) {
+		const parent = editor.dom.parentElement;
+		editor?.destroy();
```

Comment:
No need for the chaining to be optional AFAICS

```suggestion
		editor.destroy();
```